datamover/pkg/drivers/lifecycle/crosscloud_transition.go

```diff
@@ -67,11 +68,11 @@ func copyObj(ctx context.Context, obj *osdss3.Object, src *BackendInfo, dest *Ba
 		log.Logf("the transition of object[%s] is in-progress\n", obj.ObjectKey)
 		return errors.New(DMERR_TransitionInprogress)
 	}
-
+	var job *model.Job
```

Comment:
The total capacity of the whole job is used when you update progress in the function of progress, but it's value is assigned nowhere,  that means there must be 'division by zero' error.

Reply:
BTW, this file is for lifecycle, not migration, lifecycle do not need progress.

Reply:
Since crosscloud_transition also use same function which is defined for migration, I have added a dummy job here. To handle the "division by zero" I will add some filters in progress calculation to skip the calculation for lifecycle related process.

Reply:
Thanks for your update, I tested it follow the steps I used before, and it seems it works, and when I checked the log, I found those routines rarely call progress() at the same time, so I made a simulation test, the step is as below:
1. Change simuRoutines to be 60;
2. Comment out those codes that used to upload and download from clouds in the function of MoveObj to simulate that several goroutines calling the function of progress() simultaneously.
And I got the logs as below:
![log](https://user-images.githubusercontent.com/38932432/64850227-986fda00-d647-11e9-93c1-37981b9bb857.png)
The log shows before uploading, the progress already reached 91%, from the point of my view, it's not reasonable.

P.S.  I cannot access google drive currently, so if you need the test code and logs, please give me an email, I'll send to you. Thanks.

Reply:
And for the problem, that big object cannot upload, I'll check it later.

Reply:
Hi, for the crash you showed above, I think it's because bad abnormal handling, I created an issue [#183 ](https://github.com/opensds/multi-cloud/issues/183), and a PR [#184](https://github.com/opensds/multi-cloud/pull/184) to fix it.
And I think you probably encountered some environment exceptions, cause I can upload object with multipart upload successfully, even without PR#184:
![dashboard - 1](https://user-images.githubusercontent.com/38932432/64920222-8167ee00-d7e7-11e9-8e33-3a54e8e41735.png)
![log - 1](https://user-images.githubusercontent.com/38932432/64920223-85940b80-d7e7-11e9-8cfc-c757c9218d84.png)

Reply:
> Thanks for your update, I tested it follow the steps I used before, and it seems it works, and when I checked the log, I found those routines rarely call progress() at the same time, so I made a simulation test, the step is as below:
> 
> 1. Change simuRoutines to be 60;
> 2. Comment out those codes that used to upload and download from clouds in the function of MoveObj to simulate that several goroutines calling the function of progress() simultaneously.
>    And I got the logs as below:
>    ![log](https://user-images.githubusercontent.com/38932432/64850227-986fda00-d647-11e9-93c1-37981b9bb857.png)
>    The log shows before uploading, the progress already reached 91%, from the point of my view, it's not reasonable.
> 
> P.S. I cannot access google drive currently, so if you need the test code and logs, please give me an email, I'll send to you. Thanks.

@sfzeng Please share full logs if possible to roshan.wanaskar@click2cloud.net

Reply:
Hi @Click2Cloud-Gamma, sorry for late reply, the log is sent, please check. Thanks.